Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -33,98 +33,74 @@ public struct HTTPClientResponse: Sendable {
     /// The body of this HTTP response.
     public var body: Body
 
-    /// A representation of the response body for an HTTP response.
-    ///
-    /// The body is streamed as an `AsyncSequence` of `ByteBuffer`, where each `ByteBuffer` contains
-    /// an arbitrarily large chunk of data. The boundaries between `ByteBuffer` objects in the sequence
-    /// are entirely synthetic and have no semantic meaning.
-    public struct Body: Sendable {
-        private let bag: Transaction
-        private let reference: ResponseRef
-
-        fileprivate init(_ transaction: Transaction) {
-            self.bag = transaction
-            self.reference = ResponseRef(transaction: transaction)
-        }
-    }
-
     init(
         bag: Transaction,
         version: HTTPVersion,
         status: HTTPResponseStatus,
         headers: HTTPHeaders
     ) {
-        self.body = Body(bag)
         self.version = version
         self.status = status
         self.headers = headers
+        self.body = Body(TransactionBody(bag))
     }
-}
-
-@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-extension HTTPClientResponse.Body: AsyncSequence {
-    public typealias Element = AsyncIterator.Element
-
-    public struct AsyncIterator: AsyncIteratorProtocol {
-        private let stream: IteratorStream
-
-        fileprivate init(stream: IteratorStream) {
-            self.stream = stream
-        }
-
-        public mutating func next() async throws -> ByteBuffer? {
-            try await self.stream.next()
-        }
-    }
-
-    public func makeAsyncIterator() -> AsyncIterator {
-        AsyncIterator(stream: IteratorStream(bag: self.bag))
+    
+    @inlinable public init(){
```

Comment:
```suggestion
    @inlinable public init() {
```

Reply:
Is there any reason not to make this a memberwise initializer with these defaults already provided?

Reply:
We eventually want to add support for trailing headers and this would require us to add another `init` for it if we want to allow trailing headers in the init too. 
However, I'm fine with adding a member wise init already today if you are too. We aren't likely to add any more properties to this struct (famous last words) and one additional init in the future isn't too bad.